### PR TITLE
Add PR review comment command

### PR DESCRIPTION
**Pull request by reakaleek**
## What does this pull request do?

Adds a workflow that runs the full matrix which is triggered from a PR review comment with the content `/test matrix`

### CODE REVIEW

**Review by basepi — DISMISSED** [review]

**Review by reakaleek — COMMENTED** [review]
/test matrix

File: .github/workflows/README.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ Once a PR has been opened then there are two different ways you can trigger buil
 
 1. Commit based
 1. UI based, any Elasticians can force a build through the GitHub UI
+1. PR review comment based, any Elasticians can force a full matrix test run through a PR review comment with the following syntax: `/test matrix`.
```
Comment (v1v) [inline .github/workflows/README.md:42]: I think I was the one who introduced the `Elasticians` term in a few places, so let's use `Elastic employees` so it's clear for the rest of the OSS community 🙏 

```suggestion
1. PR review comment-based, any Elastic employees can force a full matrix test run through a PR review comment with the following syntax: `/test matrix`.
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -32,7 +41,9 @@ jobs:
       data: ${{ steps.split.outputs.data }}
       chunks: ${{ steps.split.outputs.chunks }}
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ inputs.ref || github.ref }}
```
Comment (v1v) [inline .github/workflows/test.yml:46]: neat!

**Review by v1v — APPROVED** [review]
just a minor comment, otherwise LGTM

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -1,7 +1,16 @@
...
 on:
-  workflow_call: ~
+  workflow_call:
+    inputs:
+      full-matrix:
+        description: "Run the full matrix"
+        required: true
+        type: boolean
+      ref:
+        description: "The git ref of elastic/apm-agent-python to build the image from. (Default: default branch)"
```
Comment (reakaleek) [inline .github/workflows/test.yml:11]:
```suggestion
        description: "The git ref of elastic/apm-agent-python to run test workflow from."
```

**Review by v1v — APPROVED** [review]

**Review by v1v — COMMENTED** [review]
/test matrix

**Review by basepi — APPROVED** [review]

**Pull request merged**